src/PackageManagement.UI/Models/SearchResultPackageMetadata.cs

```diff
@@ -23,14 +25,59 @@ public class SearchResultPackageMetadata : INotifyPropertyChanged
 
         public string Summary { get; set; }
 
+        private bool StatusProviderRun { get; set; }
+
         public PackageStatus Status
         {
-            get { return _status; }
+            get
+            {
+                if (!StatusProviderRun)
```

Comment:
After thinking about it, the current code is correct and using IsValueCreated can cause a race condition or missing an update if someone else upstream have touched the Lazy